### PR TITLE
issue-424/add jest to project 

### LINKED ISSUE #424: Add Jest to project
### Description
Jest should be implemented to allow unit testing of functions, as well as potentially testing components.

This should be done AFTER cypress is removed while implementing Playwright.

### Requirements
- [ ] Add Jest to the project
  - [ ] Configure so that it uses typescript
  - [ ] Jest test runner ignores any integration tests. 
- [ ] Write one unit test for a util function in order to validate this environment works (see `getTaskStatus.ts`).
  - [ ] Use a util function which uses absolute imports to make sure that works with the jest ts setup.

### PR DESCRIPTION
**Pull request by djbusstop**
Resolves #424

Changes:
* Adds jest and `@testing-library/react` to project and configure
  * Create custom `render` function which implements required providers for testing components
* Update README to include Jest testing instructions
* Test for `CampaignActionButtons` to test public page button navigates to correct path (validate react implementation of jest works)
* Test for `stringUtils`, `stringToBool` (validate non-react implementation of jest)
* Fixes small ts errors in `TimeFrame` that arose due to installing `ts-jest`
* Update CI
  * Use `actions/setup-node@v2` instead of `v1`, which supports yarn dependency caching
  * Lint action no longer checks types
  * New action to check types
  * New action to run jest unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,28 @@ This will also build the next.js application. You can save time and not rebuild
...
 ```
 
+### Unit tests
+
+Unit tests are run on functions and on components when we don't need to test component integrations. We use [Jest](https://jestjs.io/docs/getting-started) with [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/). To run unit tests:
+
+```
+yarn test // To run tests once.
+
+yarn test --verbose // Include this flat to always include console.log output.
```
Comment (richardolsson) [inline README.md:52]: Should _flat_ be _flag_?

File: README.md [hunk 2]
```diff
@@ -42,6 +42,28 @@ This will also build the next.js application. You can save time and not rebuild
...
+
+We override the default `render()` function from `@testing-library/react` in the file `src/test-utils/index.ts` in order to automatically apply context providers. All `@testing-library/react` functions are exposed from the custom `test-utils` module, so import from there instead of `@testing-library/react`.
+
+```js
+import { render, fireEvent } from "test-utils";
+
+/// Then use render as is documented in the docs for @testing-library/react.
+```
+
+The `react-i18n` setup in tests does not render the text in the components and instead **renders the i18n string id**. When attempting to target strings in tests, search for the id that you expect, not the translated text.
```
Comment (richardolsson) [inline README.md:65]: Is `react-i18n` in fact referring to the `react-intl` library?

**Review by richardolsson — CHANGES_REQUESTED** [review]
I think there are some spelling mistakes in the README. Other than that this is absolutely amazing. I look forward so much to being able to write simple component and unit tests in TS only, and running them as fast as these are running. 🙌

File: README.md (hunk 1 above)
Comment (djbusstop) [reply README.md:52]: aye it should :sob:

**Pull request merged**